### PR TITLE
clear media answer

### PR DESCRIPTION
**Pull request by Robert-Costello**
## Technical Summary

jira ticket: https://dimagi-dev.atlassian.net/browse/USH-3582
spec doc: https://docs.google.com/document/d/1lRYWe-IRUWN7OTYK9mVvGYhPoy_Wo85NnF0dooSw1rY/edit#heading=h.d7s2sycdfk92
hq PR: https://github.com/dimagi/commcare-hq/pull/33443

### Safety story

changes restricted to feature flag
tested locally and on staging

### QA Plan

no QA planned

### Special deploy instructions

- [x] This PR can be deployed after merge with no further considerations.

### Rollback instructions

If this PR is reverted the related hq PR should also be reverted, as `clear_answer` requests will fail.

### Review

- [x] The set of people pinged as reviewers is appropriate for the level of risk of the change.

### CODE REVIEW

**Review by shubham1g5 — CHANGES_REQUESTED** [review]
Code looks great, but need tests!

File: src/main/java/org/commcare/formplayer/application/FormController.java [hunk 1]
```diff
@@ -193,6 +204,11 @@ private FormEntryResponseBean saveAnswer(AnswerQuestionRequestBean answerQuestio
         String fileId = null;
         Path mediaDirPath = formEntrySession.getMediaDirectoryPath(restoreFactory.getDomain(),
                 restoreFactory.getUsername(), restoreFactory.getAsUsername(), storageFactory.getAppId());
+
+        if (clear) {
```
Comment (MartinRiese) [inline src/main/java/org/commcare/formplayer/application/FormController.java:208]: if `clear` is `@Nullable` this could cause a `NullPointerException`

**Review by MartinRiese — CHANGES_REQUESTED** [review]

File: src/main/java/org/commcare/formplayer/application/FormController.java (hunk 1 above)
Comment (Robert-Costello) [reply src/main/java/org/commcare/formplayer/application/FormController.java]: handled here https://github.com/dimagi/formplayer/pull/1459/commits/9af5cf36ff32d6e670942af67904d596f29d235e

Comment (shubham1g5) [reply src/main/java/org/commcare/formplayer/application/FormController.java]: booleans doesn't need to `@Nullable`, they have a definite value of `true` or `false`

Comment (Robert-Costello) [reply src/main/java/org/commcare/formplayer/application/FormController.java]: ok, so we'll just keep the `if (cear)` check and take away the `@Nullable' annotation.

Comment (MartinRiese) [reply src/main/java/org/commcare/formplayer/application/FormController.java:207]: @shubham1g5 `boolean` is `true` or `false` but `Boolean` can be `null`.

Comment (shubham1g5) [reply src/main/java/org/commcare/formplayer/application/FormController.java:207]: that is right, think we can simply use the primitive `boolean` here. 

Even if we use `Boolean`, they can't be null in this context of code as we are always passing a non null value.  So it seems ok to remove the `@Nullable` annotation and not do the null handling.

File: src/main/java/org/commcare/formplayer/application/FormController.java [hunk 2]
```diff
@@ -170,11 +170,22 @@ public FormEntryResponseBean answerMediaQuestion(
...
             throws Exception {
-        return saveAnswer(answerQuestionBean, file);
+        return saveAnswer(answerQuestionBean, file, false);
+    }
+
+    @RequestMapping(value = Constants.URL_CLEAR_ANSWER, method = RequestMethod.POST)
+    @UserLock
+    @UserRestore
+    @ConfigureStorageFromSession
+    public FormEntryResponseBean removeAnswer(
```
Comment (shubham1g5) [inline src/main/java/org/commcare/formplayer/application/FormController.java:180]: nit: rename to `clearAnswer` to be consistent with the api endpoint

**Review by shubham1g5 — CHANGES_REQUESTED** [review]

File: src/test/java/org/commcare/formplayer/tests/MediaCaptureTest.kt [hunk 3]
```diff
@@ -235,6 +236,19 @@ class MediaCaptureTest {
...
+    fun testClearMedia() {
+        val formResponse = startImageCaptureForm()
+        val imageResponse = saveImage(formResponse, "media/valid_image.jpg", "valid_image.jpg")
+
+        var expectedFilePath = getExpectedMediaPath(formResponse.session_id, imageResponse)
+        val originalSavedFile = expectedFilePath.toFile()
+        assertTrue("Could not find saved file on the filesystem", originalSavedFile.exists())
+
+        clearImage(formResponse)
+        assertFalse("Could not remove file from the filesystem", originalSavedFile.exists())
```
[the remark this answers is not in the document]
Comment (Robert-Costello) [reply src/test/java/org/commcare/formplayer/tests/MediaCaptureTest.kt:249]: https://github.com/dimagi/formplayer/pull/1459/commits/2baf39615e19de5bd8b81fa6fa00a095b2e04561

**Review by shubham1g5 — APPROVED** [review]
Good to go 💯

**Pull request merged**